### PR TITLE
Added tests for the task files

### PR DESCRIPTION
**Pull request by hiyadav**

### CODE REVIEW

File: Tasks/DotNetCoreInstallerV1/versioninstaller.ts [hunk 1]
```diff
@@ -22,64 +22,60 @@ export class VersionInstaller {
...
-                    tl.debug(tl.loc("CopyingFilesIntoPath", this.installationPath));
-                    var filesToBeCopied = allRootLevelEnteriesInDir.filter(path => !fs.lstatSync(path).isDirectory());
-                    filesToBeCopied.forEach((filePath) => {
-                        tl.cp(filePath, this.installationPath, "-f", false);
-                    });
-                }
-            }
-            catch (ex) {
-                tl.warning(tl.loc("FailedToCopyTopLevelFiles", this.installationPath, JSON.stringify(ex)));
+        return new Promise<void>((resolve, reject) => {
```
Comment (sachinma) [inline Tasks/DotNetCoreInstallerV1/versioninstaller.ts]: any reason for changing the pattern from await to using promise chain

File: Tasks/DotNetCoreInstallerV1/versioninstaller.ts [hunk 2]
```diff
@@ -22,64 +22,60 @@ export class VersionInstaller {
...
-        //todo: in case of failure due to: unable to acquire lock or timeout, 3 retries to install.
+            let version = versionInfo.getVersion();
+
+            toolLib.downloadTool(downloadUrl)
+                .then((downloadPath) => {
+                    // Extract
+                    console.log(tl.loc("ExtractingPackage", downloadPath));
+                    let extPathPromise: Promise<string> = tl.osType().match(/^Win/) ? toolLib.extractZip(downloadPath) : toolLib.extractTar(downloadPath);
+
+                    extPathPromise
```
Comment (sachinma) [inline Tasks/DotNetCoreInstallerV1/versioninstaller.ts]: try to keep a flattened promise chain

File: Tasks/DotNetCoreInstallerV1/Tests/versionInstallerIsVersionInstalledTests.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    setResourcePath: function (path) { return; }
+});
+
+import { VersionInstaller } from "../versioninstaller";
+let versionInstaller = new VersionInstaller("sdk", installationPath);
+
+if (process.env["__case__"] == "nonexplicit") {
+    let throwcount = 0;
+    try {
+        versionInstaller.isVersionInstalled("")
```
Comment (sachinma) [inline Tasks/DotNetCoreInstallerV1/Tests/versionInstallerIsVersionInstalledTests.ts]: Why can't this be in the caller and needs to be in this file?

Comment (sachinma) [reply Tasks/DotNetCoreInstallerV1/Tests/versionInstallerIsVersionInstalledTests.ts]: I thought this file only contains the mock

File: Tasks/DotNetCoreInstallerV1/versioninstaller.ts (hunk 1 above)
Comment (hiyadav) [reply Tasks/DotNetCoreInstallerV1/versioninstaller.ts]: It required use of a lot of individual try catch, was looking clumsy to me, hence changed to promise chained code.

File: Tasks/DotNetCoreInstallerV1/Tests/versionInstallerIsVersionInstalledTests.ts (hunk 3 above)
Comment (hiyadav) [reply Tasks/DotNetCoreInstallerV1/Tests/versionInstallerIsVersionInstalledTests.ts]: This was the main L0 test file only focuses on validating the results using assertions and while this individual test files take care of invoking the function to be tested.
It helped in keeping a clean and maintainable L0 test file with code reuse via individual test files.

File: Tasks/DotNetCoreInstallerV1/versioninstaller.ts (hunk 2 above)
Comment (hiyadav) [reply Tasks/DotNetCoreInstallerV1/versioninstaller.ts]: okay

File: Tasks/DotNetCoreInstallerV1/versioninstaller.ts (hunk 1 above)
Comment (hiyadav) [reply Tasks/DotNetCoreInstallerV1/versioninstaller.ts]: reverted to use await rather

File: Tasks/DotNetCoreInstallerV1/versioninstaller.ts (hunk 2 above)
Comment (hiyadav) [reply Tasks/DotNetCoreInstallerV1/versioninstaller.ts]: removed and used await

**Pull request merged**